c14550855.lua

```diff
@@ -12,8 +12,14 @@ end
 function c14550855.filter(c)
 	return c:IsSetCard(0xb) and c:IsAbleToGrave()
 end
+function c14550855.fselect(g,tp)
+	Duel.SetSelectedCard(g)
+	return Duel.CheckDiscardHand(tp,nil,0,REASON_DISCARD+REASON_EFFECT)
+end
 function c14550855.target(e,tp,eg,ep,ev,re,r,rp,chk)
-	if chk==0 then return Duel.GetFieldGroupCount(tp,LOCATION_HAND,0)>0
+	local g=Duel.GetDiscardHand(tp,REASON_DISCARD+REASON_EFFECT)
+	g:RemoveCard(e:GetHandler())
+	if chk==0 then return g:CheckSubGroup(c14550855.fselect,1,1,tp)
```

Comment:
why not just `Duel.CheckDiscardHand(tp,nil,1,REASON_DISCARD+REASON_EFFECT,e:GetHandler())`

Reply:
if `EFFECT_DISCARD_HAND_CHANGE`'s value is 0, `Duel.CheckDiscardHand` return true.
but, _Star Blast_'s FAQ is
> http://yugioh-wiki.net/index.php?%A1%D4%A5%B9%A5%BF%A1%BC%A1%A6%A5%D6%A5%E9%A5%B9%A5%C8%A1%D5#faq
> Ｑ：《魔力倹約術》でライフコストを支払わないことはできますか？
> Ａ：この場合、ライフポイントを払う事ができなくなり、このカードを発動する事自体できなくなります。(08/11/16)
so same card(e.g. _Linkslayer_) should use `Duel.SetSelectedCard`.

Reply:
I see, the situation is similar to _Destiny HERO - Diamond Dude_ cannot use _Metamorphosis_

Reply:
https://github.com/Fluorohydride/ygopro-scripts/pull/1199/commits/720b1e9601c817b18097e6d9c67ca0e48d048544 used `Duel.CheckDiscardHand`.